### DISCUSSION TITLE
How to utilise Apollo Server with Prisma 2?

### DISCUSSION THREAD

**Discussion by TheoMer** [prisma/prisma#3322] [Q&A] [answered]
"nexus": "^0.26.0-next.3",
"nexus-plugin-prisma": "^0.16.1",
"ts-node": "^8.10.2",
"typescript": "^3.9.6",
"ts-node-dev": "^1.0.0-pre.50"
"apollo-server-express": "^2.9.14"

So I'm following verbatum (graphql.ts/nexus.graphql) the [prisma-binding to Nexus](path (https://www.prisma.io/docs/guides/upgrade-guides/upgrade-from-prisma-1/upgrading-prisma-binding-to-nexus#3-migrate-graphql-operations)) documentation to migrate from Prisma 1 to 2, but wish to utilise my existing Apollo Server.

The [Old to new Nexus](https://www.prisma.io/docs/guides/upgrade-guides/upgrade-from-prisma-1/upgrading-nexus-prisma-to-nexus#21-remove-makeprismaschema) documentation states:

> With the new Nexus Framework, you don't need to create your GraphQLSchema instance manually any more as was done with makeSchema or makePrismaSchema before.

So what specific steps, in my instance, do I need to implement on Apollo Server to get it to work?

My current Apollo Server (Prisma 1 implementation) files are as follows:

1. index.ts - https://gist.github.com/TheoMer/f931e0528efb878333d10a079cec1421
2. createServer.ts - https://gist.github.com/TheoMer/a53e9734a5f3cb16dbaf8f1b1c653286
3. db.ts - https://gist.github.com/TheoMer/fb0deb45805524b057583e92cf516db5

**Comment by nikolasburk** [comment, marked as answer]
Hey @TheoMer, I believe Nexus currently has [express-graphql as a built-in GraphQL server](https://nexusjs.org/api/nexus/server#express) and believe it's not swappable to Apollo Server, unfortunately. Feel free to [open up an issue for this in the graphql-nexus repo](https://github.com/graphql-nexus/nexus/issues/new).

**Reply by TheoMer** [reply]
Hey @nikolasburk, many thanks for the heads up. My confusion lay in the difference between nexus and @nexus/schema.

I'll revert to @nexus/schema for Apollo usage.

**Reply by macrozone** [reply]
@TheoMer actually nexus uses Apollo Server and you can configure it as far as i know.

it comes with its own server, which i think is handy if you have an api-only project.

in our case we often use it with nextjs, where you can use the server-less endpoint, which also uses apollo-server, but no http server (it uses nextjs's server in this case)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!

**Comment by macrozone** [comment]
anyone stumbling over this:

if you want to create a graphql api based on prisma, I recommend pothos: https://pothos-graphql.dev/docs/plugins/prisma

it has an amazing DX and improves over nexus
